pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1943,6 +1943,27 @@ self: super: {
       )
     ];
 
+  # Pandoc 3.5 improves the quality of PDF rendering in Quarto >=1.6.30.
+  # https://github.com/NixOS/nixpkgs/pull/349683
+  pandoc-cli_3_5 = super.pandoc-cli_3_5.overrideScope (
+    self: super: {
+      doclayout = self.doclayout_0_5;
+      pandoc = self.pandoc_3_5;
+      pandoc-lua-engine = self.pandoc-lua-engine_0_3_3.overrideScope (
+        self: super: {
+          hslua-module-doclayout = self.hslua-module-doclayout_1_2_0;
+          lpeg = self.lpeg_1_1_0;
+        }
+      );
```

Comment:
afaict you can flatten this. Just override hslua-module-doclayout in the outer overrideScope and drop the inner overrideScope.